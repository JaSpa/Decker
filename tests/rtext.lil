
# test suite for the rtext interface

rt1: insert
	font:"menu" ,""          ,"mono",""
	arg :""     ,""          ,""    ,""
	text:"Hello","how about ","some"," salsa?"
into 0
rt2: insert
	text:" never ","mind."
	arg :""       ,"the_brain_page"
	font:""       ,""
into 0

show[rtext.end]
show[rtext.len[rt1]]
show[rtext.len[table ()]]
show[rtext.make["simple"]]
show[rtext.make["more" "menu" 23]]

show[rtext.get[rt1  6]]
show[rtext.get[rt1 18]]
show[rtext.get[rt1 -5]]
show[rtext.get[rt1 1000]]

show[rtext.cat[insert text:"single" into 0]]  # cast
show[rtext.cat[rt1 rt2]]                      # catenate
show[rtext.cat["one" rt2 "another"]]          # catenate w/ strings

show[rtext.string[rt1 4,19]]
show[rtext.span[rt1 8,12]]                   # onechunk
show[rtext.span[rt1 4,19]]                   # multichunk
show[rtext.span[rtext.make[""] 0,0        ]] # empty start
show[rtext.span[rtext.make[""] 0,rtext.end]] # empty end
