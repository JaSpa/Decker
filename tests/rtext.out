2147483647
26
0
+----------+------+-----+
| text     | font | arg |
+----------+------+-----+
| "simple" | ""   | ""  |
+----------+------+-----+
+--------+--------+------+
| text   | font   | arg  |
+--------+--------+------+
| "more" | "menu" | "23" |
+--------+--------+------+
1
2
0
-1
+----------+------+-----+
| text     | font | arg |
+----------+------+-----+
| "single" | ""   | ""  |
+----------+------+-----+
+------------------+--------+------------------+
| text             | font   | arg              |
+------------------+--------+------------------+
| "Hello"          | "menu" | ""               |
| "how about "     | ""     | ""               |
| "some"           | "mono" | ""               |
| " salsa? never " | ""     | ""               |
| "mind."          | ""     | "the_brain_page" |
+------------------+--------+------------------+
+--------------+------+------------------+
| text         | font | arg              |
+--------------+------+------------------+
| "one never " | ""   | ""               |
| "mind."      | ""   | "the_brain_page" |
| "another"    | ""   | ""               |
+--------------+------+------------------+
"ohow about some"
+------+-----+--------+
| font | arg | text   |
+------+-----+--------+
| ""   | ""  | " abo" |
+------+-----+--------+
+--------+-----+--------------+
| font   | arg | text         |
+--------+-----+--------------+
| "menu" | ""  | "o"          |
| ""     | ""  | "how about " |
| "mono" | ""  | "some"       |
+--------+-----+--------------+
+------+------+-----+
| text | font | arg |
+------+------+-----+
+------+------+-----+
| text | font | arg |
+------+------+-----+
()
()
()
+----------------+------+-----+
| text           | font | arg |
+----------------+------+-----+
| "first|"       | "f0" | ""  |
| "second|third" | "f1" | ""  |
| "|fourth"      | "f2" | ""  |
+----------------+------+-----+
--d1--
+---------+------+-----+
| text    | font | arg |
+---------+------+-----+
| "first" | "f0" | ""  |
+---------+------+-----+
+----------+------+-----+
| text     | font | arg |
+----------+------+-----+
| "second" | "f1" | ""  |
+----------+------+-----+
+---------+------+-----+
| text    | font | arg |
+---------+------+-----+
| "third" | "f1" | ""  |
+---------+------+-----+
+----------+------+-----+
| text     | font | arg |
+----------+------+-----+
| "fourth" | "f2" | ""  |
+----------+------+-----+
--d2--
+-------------+------+-----+
| text        | font | arg |
+-------------+------+-----+
| "first one" | "f1" | ""  |
+-------------+------+-----+
+----------+------+-----+
| text     | font | arg |
+----------+------+-----+
| "second" | "f1" | ""  |
| " one"   | "f2" | ""  |
+----------+------+-----+
+-------------+------+-----+
| text        | font | arg |
+-------------+------+-----+
| "third one" | "f3" | ""  |
+-------------+------+-----+
+------+------+-----+
| text | font | arg |
+------+------+-----+
1
1
+---------------+--------+-----+
| text          | font   | arg |
+---------------+--------+-----+
| "Hello"       | "menu" | ""  |
| "how about "  | ""     | ""  |
| "some"        | "mono" | ""  |
| " pineapple?" | ""     | ""  |
+---------------+--------+-----+
+--------------+--------+-----+
| text         | font   | arg |
+--------------+--------+-----+
| "Hello"      | "menu" | ""  |
| "how about " | ""     | ""  |
| "some"       | "mono" | ""  |
| " sal{C}"    | ""     | ""  |
+--------------+--------+-----+
+--------------+--------+-----+
| text         | font   | arg |
+--------------+--------+-----+
| "He"         | "menu" | ""  |
| "{L}{L}"     | ""     | ""  |
| "o"          | "menu" | ""  |
| "how about " | ""     | ""  |
| "some"       | "mono" | ""  |
| " sa{L}sa?"  | ""     | ""  |
+--------------+--------+-----+
+---------------+--------+-----+
| text          | font   | arg |
+---------------+--------+-----+
| "?!?how ?!? " | ""     | ""  |
| "some"        | "mono" | ""  |
| " salsa?"     | ""     | ""  |
+---------------+--------+-----+
+---------------+--------+-----+
| text          | font   | arg |
+---------------+--------+-----+
| "{A}how {B} " | ""     | ""  |
| "some"        | "mono" | ""  |
| " salsa?"     | ""     | ""  |
+---------------+--------+-----+
+-----------+--------+--------------------+
| text      | font   | arg                |
+-----------+--------+--------------------+
| "Hello"   | "menu" | ""                 |
| "how "    | ""     | ""                 |
| "ABOUT"   | "menu" | "http://about.com" |
| " "       | ""     | ""                 |
| "some"    | "mono" | ""                 |
| " salsa?" | ""     | ""                 |
+-----------+--------+--------------------+
